diff --git a/node_modules/react-native-vision-camera/android/src/main/java/com/mrousavy/camera/core/extensions/ImageCapture+takePicture.kt b/node_modules/react-native-vision-camera/android/src/main/java/com/mrousavy/camera/core/extensions/ImageCapture+takePicture.kt
index a5f8957..17cc606 100644
--- a/node_modules/react-native-vision-camera/android/src/main/java/com/mrousavy/camera/core/extensions/ImageCapture+takePicture.kt
+++ b/node_modules/react-native-vision-camera/android/src/main/java/com/mrousavy/camera/core/extensions/ImageCapture+takePicture.kt
@@ -45,12 +45,15 @@ suspend inline fun ImageCapture.takePicture(
     val outputFileOptions = outputFileOptionsBuilder.build()
 
     // Take a photo with callbacks
+    val startTime = System.currentTimeMillis()
     takePicture(
       outputFileOptions,
       executor,
       object : ImageCapture.OnImageSavedCallback {
         override fun onCaptureStarted() {
           super.onCaptureStarted()
+          val duration = System.currentTimeMillis() - startTime
+          Log.i("ImageCapture", "Capture Started Duration from start: ${duration}ms")
           if (enableShutterSound) {
             shutterSound?.play(MediaActionSound.SHUTTER_CLICK)
           }
@@ -62,6 +65,8 @@ suspend inline fun ImageCapture.takePicture(
         override fun onImageSaved(outputFileResults: ImageCapture.OutputFileResults) {
           if (continuation.isActive) {
             val info = PhotoFileInfo(file.toURI(), outputFileOptions.metadata)
+            val duration = System.currentTimeMillis() - startTime
+            Log.i("ImageCapture", "Image Saved Duration from start: ${duration}ms")
             continuation.resume(info)
           }
         }
